demo/site/src/util/siteConfig.ts

```diff
@@ -10,7 +11,7 @@ export function getHostByHeaders(headers: Headers) {
 
 export function getSiteConfigForDomain(domain: string) {
     const siteConfig = getSiteConfigs().find((siteConfig) => siteConfig.scope.domain === domain);
-    if (!siteConfig) throw new Error(`SiteConfig not found for domain ${domain}`);
+    if (!siteConfig) notFound();
```

Comment:
This should probably be handled where getSiteConfigForDomain is called, not as side effect in this helper

Reply:
@nsams also remarked this and, yes, that's true. I'll close this PR